src/main/scala/com/may/main.scala

```diff
@@ -29,15 +29,12 @@ object main extends IOApp.Simple:
     _ <- Stream
       .resource(resources)
       .flatMap { httpServer =>
-        for {
-          given Logger[IO] <- DefaultLogger.makeIo(Output.fromConsole)
-          _ <- Logger[IO].warn("Before exception") // TODO: doesn't print anything
-        } yield ()
+        Stream.eval(logger.warn("Before exception")) ++ 
```

Comment:
@hejfelix don't you know why "Before exception" is printed after "Catch me and log to console"?

<img width="1354" alt="image" src="https://github.com/emaysyuk/scala-woof-logger-issue/assets/12937626/d0031292-9c92-4cda-9d30-48528b7815c8">

Reply:
On my macbook it seems to consistently write in the "expected" order:

```shell
2024-05-21 20:14:12 [INFO ] environment=dev, application=app com.may.main$: Starting app (main.scala:28)
2024-05-21 20:14:12 [INFO ] environment=dev, application=app com.may.main$: Initializing resources (main.scala:48)
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
2024-05-21 20:14:13 [WARN ] environment=dev, application=app com.may.main$: Before exception (main.scala:32)
2024-05-21 20:14:13 [ERROR] environment=dev, application=app com.may.main$: An error occured: java.lang.RuntimeException: Catch me and log to console (main.scala:39)
```

There could be many reasons:

* stdout and stderr are not flushed to the console in the order you expect
* maybe fs2 doesn't guarantee the order of effects between each element

I think you might be running into the 1st point about when the streams are flushed to your console. Looks like you're running in `IntelliJ` or some other IDE, could you try it straight from SBT or in another IDE?

Reply:
<img width="1193" alt="image" src="https://github.com/emaysyuk/scala-woof-logger-issue/assets/12937626/2b679af3-1d09-4766-adf3-30176140bc9c">

Indeed that was IDE!

Reply:
I think it's because you need to register a logger for SLF4J (to get logs from http4s internals).

I have a branch in my fork here which registers woof as a SLF4J logging backend: https://github.com/emaysyuk/scala-woof-logger-issue/commit/fa11b62bff819dc795f0aec0a31a7dfada5679da

If this doesn't surface the error logs, then I'm not quite sure why this would happen.

Reply:
About exceptions, you should never throw exceptions in cats effect code. You should "raise" them instead, e.g.:

```scala
import cats.syntax.all.*
...
(new RuntimeException("Catch me and log to console")).raiseError
```

this works both for `Stream[F,_]` and `IO[_]` it seems. (read more here: https://typelevel.org/cats/typeclasses/applicativemonaderror.html)

`attempt` just takes e.g. `IO[A]` and turns it into `IO[Either[Throwable,A]]`. It's kind of like the cats effect equivalent of try/catch. The result will only be a `Left(...)` if program raises (or throws). 

What you're doing with `.handleWith { ... }` is perfectly fine for handling the exception!

Reply:
I recommend the typelevel discord for fast help with anything cats effect or fs2  related https://discord.com/invite/hWd4eS244g

Reply:
thank you very much for your help!